src/main.rs

```diff
@@ -1376,13 +1377,126 @@ fn parse_manpage_at_path(
+
+    // Get the number of digits in total_count
+    let padding_len = (total_count as f32).log(10.) as usize + 1;
+
+    let mut last_progress_string_length = 0;
+
+    if show_progress && !write_to_stdout {
+        println!(
+            "Parsing man pages and writing completions to {:?}",
+            output_directory.as_ref()
+        );
+    }
+
+    // As far as I can tell, we're fully supporting xz / lzma, so the checks here aren't needed
+
+    for manpage_path in paths {
+        index += 1;
+        // foo/bar/gcc.1.gz -> gcc.1.gz
+        // TODO: Expect
+        let man_file_name = manpage_path
+            .file_name()
+            .expect("Failed to get name of manpage path")
+            .to_string_lossy()
+            .to_mut()
+            .to_owned();
+
+        // gcc.1.gz -> gcc
+        cmd_name = match man_file_name.split('.').next() {
+            Some(cmd_name) => cmd_name.to_owned(),
+            None => return Err(format!("Failed to get cmd_name from {}", man_file_name)),
+        };
+
+        if show_progress {
+            // len(str(index))
+            let cur_len = (index as f32).log(10.) as usize + 1;
+            let progress_str = format!(
+                "  {} / {} : {}",
+                // str(index).rjust(padding_len)
+                format!(
+                    "{}{}",
+                    (0..(padding_len - cur_len)).fold("".to_owned(), |mut acc, _| {
+                        acc.push_str(" ");
+                        acc
+                    }),
+                    index
+                ),
+                total_count,
+                man_file_name,
+            );
+
+            // padded_progress_str = progress_str.ljust(last_progress_string_length)
+            let padded_progress_str = format!(
+                "{}{}",
+                progress_str,
+                (0..(last_progress_string_length - progress_str.len())).fold(
+                    "".to_owned(),
+                    |mut acc, _| {
+                        acc.push_str(" ");
+                        acc
+                    }
+                )
+            );
+
+            last_progress_string_length = progress_str.len();
+            // TODO: Expects
+            let stdout = std::io::stdout();
+            let mut lock = stdout.lock();
+            lock.write(format!("\r{}\r", padded_progress_str).as_bytes())
+                .expect("Failed to write to stdout");
+
+            lock.flush().expect("Failed to flush stdout");
+        }
+
+        match parse_manpage_at_path(
+            manpage_path.deref(),
+            output_directory.as_ref(),
+            deroff_only,
+            write_to_stdout,
+        ) {
+            Ok(successful) => {
+                if successful {
+                    successful_count += 1
+                }
+                // Theres no else here, i assume they just pass over it lol
+            }
+            Err(e) => {
+                // add diag, io error
```

Comment:
```suggestion
                // add_diagnostic!(format!("Cannot open {}", manpage_path));
```
We can keep like the rest as of now so I can easy move to log later.